mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +325,64 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after train epoch.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if not self.published_keys:
+            return
+
+        if self.save_last:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            if not last_ckpt:
+                assert ('Did not find last_checkpoint to be resumed.')
+            self._publish_model(runner, last_ckpt)
+
+        if self.save_best is not None:
+            best_ckpt = self.best_ckpt_path
+            if not best_ckpt:
+                assert ('Did not find best_checkpoint to be resumed.')
+            self._publish_model(runner, best_ckpt)
+
+    def _publish_model(self, runner, out_file) -> None:
+        import subprocess
+
+        import torch
+
+        checkpoint = runner.load_checkpoint(out_file)
+        ckpt_keys = list(checkpoint.keys())
+        published_keys = self.published_keys
+        if not isinstance(published_keys, list):
+            return
+        for k in ckpt_keys:
+            if k not in published_keys:
+                print_log(
+                    f'Key `{k}` will be removed because it is not in '
+                    f'save_keys. If you want to keep it, '
+                    f'please set `{k}` in published_keys',
+                    logger='current')
+                checkpoint.pop(k, None)
+        if out_file.endswith('.pth'):
+            out_file_name = out_file[:-4]
+        else:
+            out_file_name = out_file
+        tmp_out_file_name = out_file + '.pth'
+        if torch.__version__ >= '1.6':
+            torch.save(
+                checkpoint,
+                tmp_out_file_name,
+                _use_new_zipfile_serialization=False)
+        else:
+            torch.save(checkpoint, tmp_out_file_name)
+        sha = subprocess.check_output(['sha256sum',
+                                       tmp_out_file_name]).decode()
+        final_file = out_file_name + f'-{sha[:8]}.pth'
+        subprocess.Popen(['mv', tmp_out_file_name, final_file])
```

Comment:
Is it necessary to use sha to identify the checkpoint (maybe a timestamp is enough)? If it is necessary, we can use `hashlib` the calculate this, and do not need to save it as a file before.
```suggestion
        final_file = out_file_name + f'-{sha[:8]}.pth'
        subprocess.Popen(['mv', tmp_out_file_name, final_file])
```